### PR TITLE
homebrew formula for intel camera server

### PR DESCRIPTION
**Pull request by bhaney**
Made a homebrew formula for the intel realsense GRPC server, compiles and seems to work on my M1 Mac, OS Version 12.4. (though it took a long while to get the realsense hardware itself to connect to the mac)

### CODE REVIEW

**Comment by bhaney** [comment]
@tessavitabile FYI - no need to review this, but just letting you know about its existence

File: Formula/viam-rs-camera-server.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class ViamRsCameraServer < Formula
```
Comment (bhaney) [inline Formula/viam-rs-camera-server.rb:1]: Not sure what to call this. Are there issues with trademarks if we call it "ViamRealSenseCameraServer"?

File: Formula/viam-rs-camera-server.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  depends_on "protobuf"
+  depends_on "libhttpserver"
+  depends_on "opencv"
+  depends_on "librealsense"
+
+  def install
+    system "mkdir", "grpc/bin"
+    system "make", "buf"
+    system "make", "intelrealgrpcserver"
+    bin.install "intelrealgrpcserver"
```
Comment (bhaney) [inline Formula/viam-rs-camera-server.rb:22]: not sure about the name- though this is what we've always called it.

File: Formula/viam-rs-camera-server.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  desc "A Viam camera GRPC server for the Intel RealSense"
+  homepage "https://www.viam.com/"
+  url "https://github.com/viamrobotics/camera-servers/archive/refs/tags/v0.1.1.tar.gz"
+  sha256 "928cff021883e0e3740bd8732feb072315afe79bf53d928b8452d5a83ee3a67d"
+  license "Apache-2.0"
+  head "https://github.com/viamrobotics/camera-servers.git", branch: "main"
+
+  depends_on "pkg-config" => :build
+  depends_on "go" => :build
+  depends_on "openssl"
```
Comment (Otterverse) [inline Formula/viam-rs-camera-server.rb:11]: Is this explicitly needed? It builds without it, and causes some (non-fatal) conflicts with grpc when explicitly required here.

**Review by Otterverse — CHANGES_REQUESTED** [review]
A couple of minor tweaks suggested. One thing that stymied me in testing this was that brew's opencv build for linux is kinda broken. I had to install opencv from source `brew reinstall -s opencv` in order to make it work, otherwise it gave me library problems.

File: Formula/viam-rs-camera-server.rb (hunk 3 above)
Comment (bhaney) [reply Formula/viam-rs-camera-server.rb:11]: not explicitly needed, I'll remove this line

File: Formula/viam-rs-camera-server.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  depends_on "grpc"
+  depends_on "protobuf"
+  depends_on "libhttpserver"
+  depends_on "opencv"
+  depends_on "librealsense"
+
+  def install
+    system "mkdir", "grpc/bin"
```
[the remark this answers is not in the document]
Comment (bhaney) [reply Formula/viam-rs-camera-server.rb:19]: will remove

File: Formula/viam-rs-camera-server.rb (hunk 1 above)
Comment (bhaney) [reply Formula/viam-rs-camera-server.rb:1]: since the binary is called intelrealgrpcserver, I'll just call it that

**Review by Otterverse — APPROVED** [review]
Was just about to point out that the filename and classname have to match in a weird way, but you caught/fixed it. LGTM!

**Comment by Otterverse** [comment]
> made the changes, but should I just mention the `brew reinstall -s opencv` issues in the README for camera-servers?

I think that's ONLY a linux issue on certain linuxes, and generally people shouldn't be using this on linux (we have an appimage that "just works" without needing a whole ecosystem there.)

**Pull request merged**